fhd_core/visibility_manipulation/vis_flag_basic.pro

```diff
@@ -112,22 +112,27 @@ IF Keyword_Set(no_frequency_flagging) THEN BEGIN
 ENDIF ELSE freq_use=0>freq_use<1
 tile_use=0>tile_use<1
 
-;Time-based flagging
+;;; Time-based flagging
 time_use=(*obs.baseline_info).time_use
 n_time = obs.n_time
-IF Min(time_use) LE 0 THEN BEGIN
-    n_baselines=N_Elements((*obs.baseline_info).tile_A)
-    bin_offset=(*obs.baseline_info).bin_offset
-    bin_offset=[bin_offset,n_baselines]
-    time_bin=Lonarr(n_baselines)
-    FOR ti=0L,n_time-1 DO BEGIN
-        IF time_use[ti] LE 0 THEN BEGIN
-            FOR pol_i=0, n_pol-1 DO BEGIN
-                (*vis_weight_ptr[pol_i])[*, bin_offset[ti]:bin_offset[ti+1]-1] = 0
-            ENDFOR
-        ENDIF
+n_baselines=N_Elements((*obs.baseline_info).tile_A)
+bin_offset=(*obs.baseline_info).bin_offset
+bin_offset=[bin_offset,n_baselines]
+FOR ti=0L,n_time-1 DO BEGIN
+    ; Make sure the flagging of the visibilities matches the time_use array  
+    IF time_use[ti] LE 0 THEN BEGIN 
+        FOR pol_i=0, n_pol-1 DO BEGIN  
+            (*vis_weight_ptr[pol_i])[*, bin_offset[ti]:bin_offset[ti+1]-1] = 0
+        ENDFOR
+    ENDIF
+    ; Make sure the time_use array matches the flagging of the visiblities
+    FOR pol_i=0, n_pol-1 DO BEGIN
+        ; If any polarization is fully flagged, then count that time as flagged
+        time_use[ti] *= total((*vis_weight_ptr[pol_i])[*, bin_offset[ti]:bin_offset[ti+1]-1] > 0) GT 0
```

Comment:
This might need to be revisited in the case that Dara developed for HERA where antenna flagging can be per pol. I don't think we want to support different per-time flagging per pol if both pols are included, but we do want to support entirely flagging one pol of an antenna and not flagging the other pol. We can push that to another PR if we want just wanted to note the potential conflict.